commit 3f17790c2d8524c3ddc4946bd716714becf079e1
Author: Hemant Gupta <hemant.gupta@stericsson.com>
Date:   Wed Mar 28 17:09:09 2012 +0530

    Bluetooth: Use correct flags for checking HCI_SSP_ENABLED bit
    
    This patch uses the correct flags for checking the HCI_SSP_ENABLED bit.
    Without this authentication request was not being initiated.
    
    Signed-off-by: Hemant Gupta <hemant.gupta@stericsson.com>
    Acked-by: Marcel Holtmann <marcel@holtmann.org>
    Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

diff --git a/include/net/bluetooth/hci_core.h b/include/net/bluetooth/hci_core.h
index daefaac51131..8e103281a09b 100644
--- a/include/net/bluetooth/hci_core.h
+++ b/include/net/bluetooth/hci_core.h
@@ -427,7 +427,7 @@ enum {
 static inline bool hci_conn_ssp_enabled(struct hci_conn *conn)
 {
 	struct hci_dev *hdev = conn->hdev;
-	return (test_bit(HCI_SSP_ENABLED, &hdev->flags) &&
+	return (test_bit(HCI_SSP_ENABLED, &hdev->dev_flags) &&
 				test_bit(HCI_CONN_SSP_ENABLED, &conn->flags));
 }